### ISSUE TITLE
Error: Method not found: 'GoogleAuthProvider.getCredential'

### ISSUE THREAD

**Issue by Python3-8** [flutter/flutter#75605]
# The problem
I'm learning Flutter and trying to create an app with Google sign in. But I get this output is VS Code:
```
Launching lib/main.dart on iPhone 12 Pro Max in debug mode...
lib/main.dart:1
Xcode build done.                                           27.1s
Failed to build iOS app
Error output from Xcode build:
↳
    ** BUILD FAILED **
Xcode's output:
↳
    lib/auth.dart:11:56: Error: Method not found: 'GoogleAuthProvider.getCredential'.
      final AuthCredential credential = GoogleAuthProvider.getCredential(
                                                           ^^^^^^^^^^^^^
    lib/auth.dart:19:53: Error: 'currentUser' isn't a function or method and can't be invoked.
      final User currentUser = await _fbAuth.currentUser();
                                                        ^^^^...
    Command PhaseScriptExecution failed with a nonzero exit code
    note: Using new build system
    note: Building targets in parallel
    note: Planning build
    note: Constructing build description
    warning: The iOS Simulator deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 8.0, but the range of supported deployment target versions is 9.0 to 14.1.99. (in target 'AppAuth' from project 'Pods')
    warning: The iOS Simulator deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 8.0, but the range of supported deployment target versions is 9.0 to 14.1.99. (in target 'GTMAppAuth' from project 'Pods')
    warning: The iOS Simulator deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 8.0, but the range of supported deployment target versions is 9.0 to 14.1.99. (in target 'google_sign_in' from project 'Pods')
    warning: The iOS Simulator deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 8.0, but the range of supported deployment target versions is 9.0 to 14.1.99. (in target 'Flutter' from project 'Pods')
    warning: The iOS Simulator deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 8.0, but the range of supported deployment target versions is 9.0 to 14.1.99. (in target 'GoogleSignIn' from project 'Pods')
Could not build the application for the simulator.
Error launching application on iPhone 12 Pro Max.
Exited (sigterm)
```

## Steps to Reproduce

1. Run `flutter create bug`.
2. Update the files as follows:
Add Firebase to the app and enable Google sign in
Update the `lib` folder: [lib_folder.zip](https://github.com/flutter/flutter/files/5940905/lib_folder.zip)
Update the `assets` folder: [assets_folder.zip](https://github.com/flutter/flutter/files/5940914/assets_folder.zip)
Update the `pubspec.yaml` file (I've removed the comments):
```yml
name: bug
description: A new Flutter project.
publish_to: 'none'
version: 1.0.0+1

environment:
  sdk: ">=2.7.0 <3.0.0"

dependencies:
  flutter:
    sdk: flutter
  cupertino_icons: ^1.0.0
  firebase_auth: ^0.20.0+1
  firebase_core: ^0.7.0
  google_sign_in: ^4.5.9

dev_dependencies:
  flutter_test:
    sdk: flutter

flutter:

  uses-material-design: true
  assets:
    - assets/
```

**Expected results:** 
I wanted to see a "Sign in with Google" button that brings up the Google sign in page.

**Actual results:** 
I got an error while building the app.

### Logs
[flutter-run-verbose.log](https://github.com/flutter/flutter/files/5941018/flutter-run-verbose.log)

```
# output of "flutter analyze"
Analyzing learning_app...                                       

  error • The method 'getCredential' isn't defined for the type 'GoogleAuthProvider' • lib/auth.dart:11:56 • undefined_method
  error • The expression doesn't evaluate to a function, so it can't be invoked • lib/auth.dart:19:34 • invocation_of_non_function_expression
   info • Unused import: 'post.dart' • lib/login.dart:5:8 • unused_import
```
```
# output of "flutter doctor -v"
[✓] Flutter (Channel stable, 1.22.6, on Mac OS X 10.15.7 19H512 darwin-x64, locale en-US)
    • Flutter version 1.22.6 at /Users/user/AppDependencies/flutter
    • Framework revision 9b2d32b605 (2 weeks ago), 2021-01-22 14:36:39 -0800
    • Engine revision 2f0af37152
    • Dart version 2.10.5

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Users/user/Library/Android/sdk
    • Platform android-30, build-tools 30.0.3
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 12.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.1, Build version 12A7403
    • CocoaPods version 1.10.1

[!] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)

[✓] VS Code (version 1.53.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.19.0

[✓] Connected device (1 available)
    • iPhone 12 Pro Max (mobile) • EFDE6DB3-4511-4D88-AB75-AC43706CE6D1 • ios • com.apple.CoreSimulator.SimRuntime.iOS-14-1 (simulator)

! Doctor found issues in 1 category.
```
Also, there's another issue I've been facing. I've installed the Flutter and Dart plugins in Android Studio, but the Flutter CLI isn't able to detect it.

**Comment by Yayo-Arellano** [comment]
Check the latest documentation. In flutter fire website https://firebase.flutter.dev/docs/auth/social#google

To get the current user is not async anymore.  `User user = FirebaseAuth.instance.currentUser;` 

To login check this code fragment. 
```
Future<UserCredential> signInWithGoogle() async {
  // Trigger the authentication flow
  final GoogleSignInAccount googleUser = await GoogleSignIn().signIn();

  // Obtain the auth details from the request
  final GoogleSignInAuthentication googleAuth = await googleUser.authentication;

  // Create a new credential
  final GoogleAuthCredential credential = GoogleAuthProvider.credential(
    accessToken: googleAuth.accessToken,
    idToken: googleAuth.idToken,
  );

  // Once signed in, return the UserCredential
  return await FirebaseAuth.instance.signInWithCredential(credential);
}
```

**Comment by Python3-8** [comment]
Thanks, I _was_ wondering if it was different in the new version but I didn't know where to find the documentation.

**Comment by Python3-8** [comment]
I updated `auth.dart` to look like this:
```dart
import 'package:firebase_auth/firebase_auth.dart';
import 'package:google_sign_in/google_sign_in.dart';

final FirebaseAuth _fbAuth = FirebaseAuth.instance;
final GoogleSignIn _googleSignIn = GoogleSignIn();

Future<User> signInWithGoogle() async {
  // Trigger the authentication flow
  final GoogleSignInAccount googleUser = await _googleSignIn.signIn();

  // Obtain the auth details from the request
  final GoogleSignInAuthentication googleAuth = await googleUser.authentication;

  // Create a new credential
  final GoogleAuthCredential credential = GoogleAuthProvider.credential(
    accessToken: googleAuth.accessToken,
    idToken: googleAuth.idToken,
  );
  final UserCredential authResult =
      await _fbAuth.signInWithCredential(credential);
  final User user = authResult.user;
  assert(!user.isAnonymous);
  assert(await user.getIdToken() != null);
  final User currentUser = _fbAuth.currentUser;
  assert(currentUser.uid == user.uid);
  return user;
}

// final GoogleSignInAccount googleSignInAccount = await _googleSignIn.signIn();
// final GoogleSignInAuthentication googleSignInAuthentication =
//     await googleSignInAccount.authentication;
// final AuthCredential credential = GoogleAuthProvider.getCredential(
//     idToken: googleSignInAuthentication.idToken,
//     accessToken: googleSignInAuthentication.accessToken);
// final UserCredential authResult =
//     await _fbAuth.signInWithCredential(credential);
// final User user = authResult.user;
// assert(!user.isAnonymous);
// assert(await user.getIdToken() != null);
// final User currentUser = await _fbAuth.currentUser();
// assert(currentUser.uid == user.uid);
// return user;
```
And now my app crashes when I click the "Sign in with Google" button: https://user-images.githubusercontent.com/66139317/107186684-5668d600-69aa-11eb-8ed7-f276355a343b.mov

**Comment by darshankawar** [comment]
@Python3-8 
Since you are using `google_sign_in` + `firebase_auth`, then, you'll need to raise this issue in firebase dedicated repo : https://github.com/FirebaseExtended/flutterfire/issues.
Closing from here.
